### PR TITLE
Replace schemaJob with a call to management API

### PR DESCRIPTION
**Pull request by burmanm**
Fixes https://github.com/k8ssandra/k8ssandra/issues/227

### CODE REVIEW

File: pkg/reconcile/reconcilers_test.go [hunk 1]
```diff
@@ -38,6 +37,7 @@ func TestNewService(t *testing.T) {
 	assert.Equal(t, port, service.Spec.Ports[0])
 }
 
+/*
```
Comment (jeffbanks) [inline pkg/reconcile/reconcilers_test.go:40]: Dead code.

**Review by jeffbanks — CHANGES_REQUESTED** [review]

File: controllers/suite_test.go [hunk 2]
```diff
@@ -60,19 +64,33 @@ func TestAPIs(t *testing.T) {
...
+			filepath.Join("..", "test", "config", "cass-operator-crd", "bases"),
 		},
 	}
 
-	var err error
 	cfg, err = testEnv.Start()
 	Expect(err).ToNot(HaveOccurred())
 	Expect(cfg).ToNot(BeNil())
 
+	// time.Sleep(20 * time.Second)
```
[the remark this answers is not in the document]
Comment (burmanm) [reply controllers/suite_test.go:92]: Dead code.

File: pkg/reconcile/reconcilers_test.go [hunk 3]
```diff
@@ -4,7 +4,6 @@ import (
 	"testing"
```
[the remark this answers is not in the document]
Comment (burmanm) [reply pkg/reconcile/reconcilers_test.go:4]: No.

File: test/config/kind/kind_config_1_worker.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+kind: Cluster
```
[the remark this answers is not in the document]
Comment (burmanm) [reply test/config/kind/kind_config_1_worker.yaml]: reaper-operator has had different folder naming (long before k8ssandra test naming)

File: test/framework/framework.go [hunk 5]
```diff
@@ -98,6 +100,23 @@ func KustomizeAndApply(namespace, dir string) {
...
 }
 
+func WaitForCRDs(namespace string) {
+	// Wait for Reaper CRD
+	var stdout bytes.Buffer
+	kubectl := exec.Command("kubectl", "-n", namespace, "wait", "--for", "condition=established", "--timeout=60s", "crd/reapers.reaper.cassandra-reaper.io")
+	kubectl.Stdin = &stdout
+	out, err := kubectl.CombinedOutput()
+	GinkgoWriter.Write(out)
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("kubectl wait failed: %s", err))
```
[the remark this answers is not in the document]
Comment (burmanm) [reply test/framework/framework.go:110]: I don't think that matters, we're mostly checking error here for the execution code (whatever kubectl returns). So there should always be output.

File: test/config/kind/kind_config_1_worker.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeffbanks) [reply test/config/kind/kind_config_1_worker.yaml]: Fair enough.

File: pkg/reconcile/reconcilers_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffbanks) [reply pkg/reconcile/reconcilers_test.go:4]: Not that this code base is too big, but that approach can make it _interesting_ for developers when they see a `test/` folder and then later see `_test.go` files residing along side non-test files.  

Just not the approach we've been following in our other repositories for K8ssandra.  Not a show stopper for me though.

File: Makefile [hunk 6]
```diff
@@ -43,8 +43,9 @@ all: manager
 ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 test: generate fmt vet manifests
 	mkdir -p ${ENVTEST_ASSETS_DIR}
-	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/master/hack/setup-envtest.sh
-	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh && fetch_envtest_tools $(ENVTEST_ASSETS_DIR) && setup_envtest_env $(ENVTEST_ASSETS_DIR) && go test ./controllers/... ./pkg/... -coverprofile cover.out
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
```
[the remark this answers is not in the document]
Comment (jeffbanks) [reply Makefile:46]: Gotcha, didn't know the backstory here, thanks.

**Review by jeffbanks — APPROVED** [review]

**Pull request merged**